### PR TITLE
Runtime language metrics

### PR DESCRIPTION
**Pull request by maiqbal11**

### CODE REVIEW

**Comment by ColbyTresness** [comment]
@fabiocav @maiqbal11 I'm wondering if its best to have this as a single metric, host.startup.language, with a parameter detailing which language? That way we won't have to do RegEx matching on the metric name

**Comment by maiqbal11** [comment]
> @fabiocav @maiqbal11 I'm wondering if its best to have this as a single metric, host.startup.language, with a parameter detailing which language? That way we won't have to do RegEx matching on the metric name

Chatted with @brettsam briefly about this. It looks like the current way we surface metrics is to use dot separated names for what we are specifying. From my understanding, there currently doesn't seem to be a mechanism for adding a separate column for the language. Did you have something else in mind with the parameter you had mentioned?

**Comment by fabiocav** [comment]
We do have a column for the metric message, which could be used for this information. I agree that having a single metric name would be a better approach.

File: src/WebJobs.Script/Diagnostics/MetricEventNames.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public static class MetricEventNames
 
         // language worker level events
         public const string WorkerInitializeLatency = "host.startup.outofproc.{0}worker.initialize.attempt{1}.latency";
+        public const string WorkerRuntimeLanguage = "host.startup.language.{0}";
```
Comment (mathewc) [inline src/WebJobs.Script/Diagnostics/MetricEventNames.cs]: nit: HostStartupLanguage, to match naming convention for other host startup metrics. Also, move this up next to its brothers above (host.startup.latency, etc.)

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 2]
```diff
@@ -260,6 +260,11 @@ protected override async Task StartAsyncCore(CancellationToken cancellationToken
         public async Task InitializeAsync()
         {
             _stopwatch.Start();
+            if (!_environment.IsPlaceholderModeEnabled())
+            {
+                string runtimeLanguage = string.IsNullOrEmpty(_currentRuntimelanguage) ? "none" : _currentRuntimelanguage;
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHost.cs]: nit: can you fix the casing of _currentRuntimelanguage? I realize you didn't add that member :)

File: src/WebJobs.Script/Diagnostics/MetricEventNames.cs (hunk 1 above)
Comment (fabiocav) [reply src/WebJobs.Script/Diagnostics/MetricEventNames.cs]: For accuracy, the metric name should also be `host.startup.runtime`

**Comment by brettsam** [comment]
@ColbyTresness -- Opening another issue sounds good. Ideally we'd structure this data in the best way for us to easily consume it, but that'll require some changes to how we write out these metrics, which is a bigger work item. With this change, at least we'll have the data we want... even if it's tougher to extract.

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 2 above)
Comment (brettsam) [inline src/WebJobs.Script/Host/ScriptHost.cs]: This isn't telling us the number of functions in that language, right? Just whatever the app setting was?

@ColbyTresness -- Can you extract the details you want just by having this? I know @fabiocav wanted to make sure we could track mixed-language apps, which isn't technically supported, but may still happen.

Comment (maiqbal11) [reply src/WebJobs.Script/Host/ScriptHost.cs]: Right, this is just telling us what language the application was configured to run.

Comment (mathewc) [reply src/WebJobs.Script/Host/ScriptHost.cs]: This isn't a count of the number of functions for a particular runtime - it's a count of the number of times a host has started/restarted for a particular language. This metric wouldn't make sense for multi-language hosts. If we wanted to handle that, we'd instead have a function.runtime.{runtime} metric, similar to the existing function.binding.{bindingType} metric which provides counts per binding on startup.

Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptHost.cs]: I think that, particularly after some recent work with the out-of-proc workers, mixed language environments will no longer be something we need to consider, so this work.

Comment (brettsam) [reply src/WebJobs.Script/Host/ScriptHost.cs]: Right -- I want to make sure that this is what @ColbyTresness and @jeffhollan need. I thought someone mentioned wanting the count of functions-per-language (or maybe it was total-executions-per-language?). I'm getting all the conversations mixed up in my head :-) 

As long as they're good with the metric and have a way to pull what they need, this looks good to me.

Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptHost.cs]: You're remembering the conversation we had. With the more strict behavior where mixed runtimes are not supported, having this as is will also allow the function counts to be derived, so I think this is sufficient.

**Pull request merged**